crates/typst-pdf/src/outline.rs

```diff
@@ -18,7 +18,17 @@ pub(crate) fn write_outline(ctx: &mut PdfContext) -> Option<Ref> {
     // enforced in the manner shown below.
     let mut last_skipped_level = None;
     let elements = ctx.document.introspector.query(&HeadingElem::elem().select());
+
     for elem in elements.iter() {
+        if let Some(page_ranges) = &ctx.exported_pages {
+            if !page_ranges.page_included(
+                ctx.document.introspector.page(elem.location().unwrap()).get() - 1,
+            ) {
```

Comment:
Maybe we could have `includes_page` and `includes_page_index`, so that we don't need to have `-1` sprinkled randomly.